esp/bindings/http/platform/httptransport.cpp

```diff
@@ -2123,8 +2123,8 @@ int CHttpRequest::processHeaders(IMultiException *me)
 
 bool CHttpRequest::readContentToBuffer(MemoryBuffer& buffer, __int64& bytesNotRead)
 {
-    char buf[1024 + 1];
-    __int64 buflen = 1024;
+    char buf[1048576 + 1];
```

Comment:
Allocating 1MB on the stack could cause problems.

The function also has a couple of other problems:
* It is unnecessarily memcpy-ing the data - which adds up if the file is large.
* It is unnecessarily adding a null terminator onto the string that was read (which makes the code confusing).

Better is something like:

```
    constexpr size32_t readChunkSize = 0x100000;
    size32_t sizeToRead = bytesNotRead > readChunkSize ? readChunkSize: (size32_t)readChunkSize;
    size32_t prevLen = buffer.length();
    char * target = (char *)buffer.reserve(sizeToRead);
    int readlen = m_bufferedsocket->read(target, sizeToRead);
    if(readlen <= 0)
    {
        if(readlen < 0)
            DBGLOG("Failed to read from socket");
        buffer.setLength(prevLen);
        return false;
    }

    buffer.setLength(prevLen + readlen);
    bytesNotRead -= readlen;
    return true;

```

(Untested).  It reads the data directly into the buffer rather than copying it.

Reply:
This makes sense, it is a much better solution. I'd completely missed the consideration of stack size and the reserve on the buffer is slick. I'm implementing this and testing.